### PR TITLE
ES6: Migrated "assets"

### PR DESCRIPTION
**Pull request by herrmannplatz**
- [x] fix test
- [x] update comments

### CODE REVIEW

File: app/script/assets/AssetMetaDataBuilder.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+    if (this.is_video(file)) {
+      return this._build_video_metdadata(file);
+    } else if (this.is_audio(file)) {
+      return this._build_audio_metdadata(file);
+    } else if (this.is_image(file)) {
+      return this._build_image_metdadata(file);
+    } else {
+      return Promise.resolve();
+    }
```
Comment (lipis) [inline app/script/assets/AssetMetaDataBuilder.js:47]: I'm no fan of single line if statements.. but applies nicely here:

```js
if (this.is_video(file)) return this._build_video_metdadata(file);
if (this.is_audio(file)) return this._build_audio_metdadata(file);
if (this.is_image(file)) return this._build_image_metdadata(file);
return Promise.resolve();
```

File: app/script/assets/AssetService.js [hunk 2]
```diff
@@ -0,0 +1,467 @@
...
+      const xhr = new XMLHttpRequest();
+      xhr.open('POST', this.client.create_url('/assets/v3'));
+      xhr.setRequestHeader('Content-Type', `multipart/mixed; boundary=${this.BOUNDARY}`);
+      xhr.setRequestHeader('Authorization', `${this.client.access_token_type} ${this.client.access_token}`);
+      xhr.onload = function(event) {
+        if (this.status === 201) {
+          return resolve(JSON.parse(this.response));
+        } else {
+          return reject(event);
+        }
```
Comment (lipis) [inline app/script/assets/AssetService.js]:
```js
if (this.status === 201) {
  return resolve(JSON.parse(this.response));
}
return reject(event);
```

File: app/script/assets/AssetUploadFailedReason.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+ */
+
+'use strict';
+
+window.z = window.z || {};
+window.z.assets = z.assets || {};
+
+z.assets.AssetUploadFailedReason = {
+  FAILED: 1,
+  CANCELLED: 0,
```
Comment (lipis) [inline app/script/assets/AssetUploadFailedReason.js]: alphabetical :)

File: app/script/assets/AssetMetaDataBuilder.js (hunk 1 above)
Comment (bennycode) [reply app/script/assets/AssetMetaDataBuilder.js:47]: But but but... #consistency

Comment (lipis) [reply app/script/assets/AssetMetaDataBuilder.js:47]: like I said.. already.. I'm not a big fan of it..! So I won't mind if not changed :)

**Pull request merged**